pmd-core/src/main/java/net/sourceforge/pmd/AbstractConfiguration.java

```diff
@@ -194,6 +199,14 @@ public void setDefaultLanguageVersions(List<LanguageVersion> languageVersions) {
         }
     }
 
+    /**
+     * Check that it is correct to use the given language with this configuration.
+     * Throw if not.
+     */
+    protected void checkLanguageIsAcceptable(Language lang) {
```

Comment:
should we make this `abstract` and force concrete implementations to implement it, even if empty, rather than be an opt-in to do these checks?

Also, I'd explicitly add a `throws UnsupportedOperationException`. Even if unneeded (it's a RTE), it avoids any hesitation as to what exception to throw if not valid… The doc should also state that, rather than simply say "Throw if now".

Reply:
on second thought, making this abstract would be a breaking API change…